### DISCUSSION TITLE
Ability to override a model. Complex TS types usage.

### DISCUSSION THREAD

**Discussion by MatanYadaev** [prisma/prisma#12564] [Q&A]
I would like to improve my model type.

This is my Prisma model:
```prisma
model Request {
  id          Int     @id @default(autoincrement())
  user        User    @relation(fields: [userId], references: [id])
  userId      Int     @unique
  approvedAt  DateTime?
  rejectedAt  DateTime?
}
```

This is the model type Prisma generated:
```ts
export type Request = {
  id: number
  userId: number
  approvedAt: Date | null
  rejectedAt: Date | null
}
```

I would like to improve this `Request` type and to make it more specific:
```ts
type PendingRequest = {
  id: number
  userId: number
  approvedAt: null
  rejectedAt: null
}

type ApprovedRequest = {
  id: number
  userId: number
  approvedAt: Date
  rejectedAt: null
}

type RejectedRequest = {
  id: number
  userId: number
  approvedAt: null
  rejectedAt: Date
}

export type Request = PendingRequest | ApprovedRequest | RejectedRequest
```

There is a way to override model types that achieves the same results? That provides model a few different states?

**Comment by austincrim** [comment]
Hey @MatanYadaev 👋 ,

It looks like you created [an issue](https://github.com/prisma/prisma/issues/12567) for this. Further updates will take place over there 😄.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
